drydock/plugin.py

```diff
@@ -1,11 +1,129 @@
+    tutor_conf = tutor_config.load(context.root)
+
+    for i, (service, command) in enumerate(init_tasks):
+        for template in _load_jobs(tutor_conf):
+            if template['metadata']['name'] != service + '-job':
+                continue
+
+            render_command = tutor_env.render_str(tutor_conf, command)
+
+            template['metadata']['name'] = 'drydock-' + template['metadata']['name'] + '-' + str(i)
+            template['metadata']['labels'].update({
+                'app.kubernetes.io/component': 'drydock-job',
+                'drydock.io/target-service': template['metadata']['name'],
+                'drydock.io/runner-service': template['metadata']['name']
+            })
+            template['metadata']['annotations'] = {
+                'argocd.argoproj.io/sync-wave': INIT_JOBS_SYNC_WAVE + i * 2,
+                'argocd.argoproj.io/hook': 'Sync',
+                'argocd.argoproj.io/hook-delete-policy': 'HookSucceeded,BeforeHookCreation'
+            }
+
+            shell_command = ["sh", "-e", "-c"]
+            if template["spec"]["template"]["spec"]["containers"][0].get("command") == []:
+                # In some cases, we need to bypass the container entrypoint.
+                # Unfortunately, AFAIK, there is no way to do so in K8s manifests. So we mark
+                # some jobs with "command: []". For these jobs, the entrypoint becomes "sh -e -c".
+                # We do not do this for every job, because some (most) entrypoints are actually useful.
+                template["spec"]["template"]["spec"]["containers"][0]["command"] = shell_command
+                container_args = [render_command]
+            else:
+                container_args = shell_command + [render_command]
+
+            template["spec"]["template"]["spec"]["containers"][0]["args"] = container_args
+            template["spec"]["backoffLimit"] = 1
+            template["spec"]["ttlSecondsAfterFinished"] = 3600
+
+
+            yield serialize.dumps(template)
+
+
+CORE_SYNC_WAVES_ORDER: SYNC_WAVES_ORDER_ATTRS_TYPE = {
+    "drydock-upgrade-lms-job": 50,
+    "drydock-upgrade-cms-job": 51,
+    "lms-lifecycle-enabled": 100,
+    "cms-lifecycle-enabled": 100,
+    "lms-debug": 50,
+    "cms-debug": 50,
+    "ingress-debug": 200,
+    "horizontalpodautoscalers:all": 150
+}
+
+
+@SYNC_WAVES_ORDER.add()
+def _add_core_sync_waves_order(sync_waves_config: SYNC_WAVES_ORDER_ATTRS_TYPE) -> SYNC_WAVES_ORDER_ATTRS_TYPE:
+    sync_waves_config.update(CORE_SYNC_WAVES_ORDER)
+    return sync_waves_config
+
+
+@functools.lru_cache(maxsize=None)
+def get_sync_waves_order() -> SYNC_WAVES_ORDER_ATTRS_TYPE:
+    """
+    This function is cached for performance.
+    """
+    return SYNC_WAVES_ORDER.apply({})
+
+
+@tutor_hooks.Actions.PLUGIN_LOADED.add()
+def _clear_sync_waves_order_cache(_name: str) -> None:
+    """
+    Don't forget to clear cache, or we'll have some strange surprises...
+    """
+    get_sync_waves_order.cache_clear()
```

Comment:
can we can replace this with https://github.com/overhangio/tutor/blob/f211b982fd8d970a538eb8452e16259a5eaae92b/tutor/hooks/__init__.py#L16 ?